Core/PARStore.m

```diff
@@ -949,6 +950,85 @@ - (void)setEntriesFromDictionary:(NSDictionary *)dictionary
      }];
 }
 
+- (void)appendChanges:(NSArray *)changes forDeviceIdentifier:(NSString *)deviceIdentifier
+{
+    // Model and PSC
+    NSManagedObjectModel *mom = [PARStore managedObjectModel];
+    NSPersistentStoreCoordinator *psc = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:mom];
+    
+    // Persistent Store
+    NSError *error = nil;
+    NSString *dirPath = [self directoryPathForDeviceIdentifier:deviceIdentifier];
+    [[NSFileManager defaultManager] createDirectoryAtPath:dirPath withIntermediateDirectories:NO attributes:nil error:NULL];
+    [self addPersistentStoreWithCoordinator:psc dirPath:dirPath readOnly:NO error:&error];
+    
+    // Context
+    NSManagedObjectContext *moc = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSPrivateQueueConcurrencyType];
+    [moc setPersistentStoreCoordinator:psc];
+    [moc setUndoManager:nil];
+
+    // Have main context observe changes made in background, and merge them.
+    id observer = [[NSNotificationCenter defaultCenter] addObserverForName:NSManagedObjectContextDidSaveNotification object:moc queue:nil usingBlock:^(NSNotification *notification) {
```

Comment:
They work as advertised. You just have to follow the rules, which is to remove the returned observer object. We are doing that, and it avoids having to add extra methods for what is quite a localised usage, so I think it is OK here.